packages/venia-concept/src/actions/checkout/__tests__/asyncActions.spec.js

```diff
@@ -0,0 +1,219 @@
+import { RestApi } from '@magento/peregrine';
+
+import { store } from 'src';
+import actions from '../actions';
+import {
+    editOrder,
+    formatAddress,
+    resetCheckout,
+    submitCart,
+    submitInput,
+    submitOrder
+} from '../asyncActions';
+
+jest.mock('src');
+
+const { dispatch, getState } = store;
+const thunkArgs = [dispatch, getState];
+const { request } = RestApi.Magento2;
+
+const address = {
+    country_id: 'US',
+    firstname: 'Veronica',
+    lastname: 'Costello',
+    street: ['6146 Honey Bluff Parkway'],
+    city: 'Calder',
+    postcode: '49628-7978',
+    region_id: 33,
+    region_code: 'MI',
+    region: 'Michigan',
+    telephone: '(555) 229-3326',
+    email: 'veronica@example.com'
+};
+
+const countries = [
+    { id: 'US', available_regions: [{ id: 33, code: 'MI', name: 'Michigan' }] }
+];
+
+beforeAll(() => {
+    getState.mockImplementation(() => ({
+        cart: { guestCartId: 'GUEST_CART_ID' },
+        directory: { countries }
+    }));
+});
+
+afterEach(() => {
+    dispatch.mockClear();
+    request.mockClear();
+});
+
+afterAll(() => {
+    getState.mockRestore();
+});
+
+test('resetCheckout() returns a thunk', () => {
+    expect(resetCheckout()).toBeInstanceOf(Function);
+});
+
+test('resetCheckout thunk returns undefined', async () => {
+    const result = await resetCheckout()(...thunkArgs);
+
+    expect(result).toBeUndefined();
+});
+
+test('resetCheckout thunk dispatches actions', async () => {
+    await resetCheckout()(...thunkArgs);
+
+    expect(dispatch).toHaveBeenNthCalledWith(1, expect.any(Function));
+    expect(dispatch).toHaveBeenNthCalledWith(2, actions.reset());
+    expect(dispatch).toHaveBeenCalledTimes(2);
+});
+
+test('editOrder() returns a thunk', () => {
+    expect(editOrder()).toBeInstanceOf(Function);
+});
+
+test('editOrder thunk returns undefined', async () => {
+    const result = await editOrder()(...thunkArgs);
```

Comment:
Slightly different pattern here than in the other places you're testing whether a thunk returns undefined. Should stay consistent for clarity's sake, because a new developer will probably need some time understanding what this is and why a thunk should return (or resolve) undefined.